### PR TITLE
Implement 'packaging' config option in Artman

### PR DESCRIPTION
**Pull request by shinfan**
This PR enables Artman to package the generated GAPIC client as a fat package that is ready to deploy and execute. (Currently this only supports Java)

Example command:
```
artman --api pubsub --language java

cd /path/to/output/google-cloud-pubsub/
./gradlew jar
```
This feature is motivated by the Java DPE hand-off process that the DPEs have been looking for a executable-ready sample application. Before this PR the process is manual.

Toolkit PR: https://github.com/googleapis/toolkit/pull/1325

### CODE REVIEW

**Comment by shinfan** [comment]
Friendly ping

File: artman/cli/main.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def parse_args(*args):
              'argument.',
     )
     parser.add_argument('--publish',
-        choices=('github', 'local', 'maven', 'noop'),
+        choices=('github', 'local', 'maven', 'noop', 'sample_app'),
         default=None,
         help='Set where to publish the code. Options are "github", "local", '
```
Comment (geigerj) [inline artman/cli/main.py]: Update `help`, please.

File: artman/tasks/publish/sample_app.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+from artman.tasks import task_base
+from artman.utils.logger import logger
+
+class _JavaSampleTask(task_base.TaskBase):
+    def execute(self, gapic_code_dir=None, grpc_code_dir=None, proto_code_dir=None):
+        
+        if grpc_code_dir:
+            self.exec_command(['cp', '-rf', grpc_code_dir, gapic_code_dir])
+        else:
+            raise Exception('GRPC package is not generated.')
```
Comment (geigerj) [inline artman/tasks/publish/sample_app.py]: Don't raise a base `Exception`.

File: artman/tasks/publish/sample_app.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+import six
+
+from artman.tasks import task_base
+from artman.utils.logger import logger
+
+class _JavaSampleTask(task_base.TaskBase):
+    def execute(self, gapic_code_dir=None, grpc_code_dir=None, proto_code_dir=None):
+        
+        if grpc_code_dir:
```
Comment (geigerj) [inline artman/tasks/publish/sample_app.py]: If you're going to raise an exception without `gapic_code_dir` or `proto_code_dir` anyway, why make them optional arguments at all?

File: artman/tasks/publish/sample_app.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import uuid
+
+import github3
+
+import six
+
+from artman.tasks import task_base
+from artman.utils.logger import logger
+
+class _JavaSampleTask(task_base.TaskBase):
```
Comment (geigerj) [inline artman/tasks/publish/sample_app.py]: Short comment explaining what this task does?

**Comment by shinfan** [comment]
The Java smoke test failed because I haven't updated the googleapis configs. It should pass once the googleapis change has been made.

**Comment by shinfan** [comment]
@lukesneeringer I answered your question in a shared design doc.

**Pull request merged**